### DISCUSSION TITLE
Feature Request: Data Cleanup Policy

### DISCUSSION THREAD

**Discussion by CallumJHays** [open-webui/open-webui#7465] [Ideas]
We are looking to deploy open-webui internally at my company to centralise AI usage for work and reduce risk of IP leakage.
Use cases include workshopping official communications such as announcements and management reviews.

Due to the sensitive nature of the chats and uploaded files, we would like to implement a data cleanup policy that includes:
- deleting chats older than a specific threshold from the database, including related entities:
   - uploaded files from the `file` table
   - uploaded files from `StorageProvider`
   - uploaded file collections in vector DB
- deleting files older than a specific threshold from the `CACHE_DIR`

I have an initial version of the feature included in our internal deployment. It uses `apscheduler` to manage a periodic cleanup job. 

I'm happy to open a PR if the feature request & implementation sounds reasonable 😺

**Comment by CallumJHays** [comment]
This feature is related to the following discussions:
- https://github.com/open-webui/open-webui/discussions/5199
- https://github.com/open-webui/open-webui/discussions/6705

**Comment by gaby** [comment]
@CallumJHays If you contribute this, it would be awesome.

**Comment by CallumJHays** [comment]
Closing in favor of https://github.com/open-webui/open-webui/discussions/12091

**Reply by CallumJHays** [reply]
But PS: (closed) PR with implementation here: https://github.com/open-webui/open-webui/pull/13396

**Comment by Classic298** [comment]
Hello everyone.

I worked on a PR related to this discussion.
Testing wanted!

https://github.com/open-webui/open-webui/pull/16520
